plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/WizardConfig.xtend

```diff
@@ -44,151 +32,82 @@ class WizardConfig extends XtextProjectConfig {
 	
 	override checkConfiguration(Issues issues) {
 		super.checkConfiguration(issues)
-		if (rootPath == null) {
-			issues.addError('The property \'rootPath\' must be set.', this)
-			return
-		}
-		if (baseName == null) {
-			issues.addError('The property \'baseName\' must be set.', this)
-			return
-		}
-		val runtimeBase = runtimeRoot?.path
-		if (!Character.isJavaIdentifierPart(runtimeBase.charAt(runtimeBase.length - 1)))
-			issues.addError('The runtime root path must end with a valid package name.', this)
-		if ((ideaEditor || webSupport) && !genericIdeSupport)
-			issues.addError('Generic IDE support must be enabled when the IDEA or web editors are enabled.', this)
+		if (rootPath === null)
+			issues.addError("The property 'rootPath' must be set", this)
+		if (baseName === null)
+			issues.addError("The property 'baseName' must be set", this)
 	}
-	
-	override initialize(Injector injector) {
-		var src = 'src'
-		var testSrc = src
-		var srcGen = 'src-gen'
-		var testSrcGen = srcGen
-		var metaInf = 'META-INF'
-		var modelGen = 'model/generated'
-		var srcWeb = 'WebRoot'
-		if (mavenLayout) {
-			src = 'src/main/java'
-			testSrc = 'src/test/java'
-			srcGen = 'src/main/xtext-gen'
-			testSrcGen = 'src/test/xtext-gen'
-			metaInf = 'src/main/resources/META-INF'
-			modelGen = 'src/main/model-gen'
-			srcWeb = 'src/main/webapp'
-		}
-		
-		if (runtimeProjectName=== null)
-			runtimeProjectName = baseName
-		if (runtimeRoot === null)
-			runtimeRoot = rootPath + '/' + runtimeProjectName
-		if (runtimeMetaInf === null)
-			runtimeMetaInf = runtimeRoot.path + '/' + metaInf
-		if (runtimeSrc === null)
-			runtimeSrc = runtimeRoot.path  + '/' + src
-		if (runtimeSrcGen === null)
-			runtimeSrcGen = runtimeRoot.path  + '/' + srcGen
-		if (runtimeManifest === null)
-			runtimeManifest = injector.getInstance(ManifestAccess)
-		if (runtimePluginXml === null)
-			runtimePluginXml = injector.getInstance(PluginXmlAccess)
-		if (runtimeModelGen === null)
-			runtimeModelGen = runtimeRoot.path + '/' + modelGen
-			
-		if (testingSupport) {
-			if (runtimeTestProjectName=== null)
-				if(mavenLayout) {
-					runtimeTestProjectName = runtimeProjectName
-				} else {
-					runtimeTestProjectName = runtimeProjectName + ".tests"
+
+	override setDefaults() {
+		super.setDefaults
+		enabledProjects.forEach [
+			if (name === null)
+				name = computeName
+			if (root === null)
+				root = computeRoot
+			if (metaInf === null)
+				metaInf = computeMetaInf
+			if (src === null)
+				src = computeSrc
+			if (srcGen === null)
+				srcGen = computeSrcGen
+			if (it instanceof BundleProjectConfig) {
+				if (createEclipseMetaData) {
+					if (manifest === null)
+						manifest = new ManifestAccess
```

Comment:
I prefer the variant that used the injector to create the manifestAccess since subclasses could be bound. This approach only allows to inject other properties into the *accesses but doesn't provide the advantage of inheritance / polymorphism. Is there a specific reason to do it like this that I don't understand yet?